packages/teams-js/src/public/location.ts

```diff
@@ -45,82 +50,61 @@ export namespace location {
+   * As of 2.0.1, please use one of the following functions:
+   * - {@link geoLocation.getCurrentLocation geoLocation.getCurrentLocation(): Promise\<Location\>} to get the current locaiton.
+   * - {@link geoLocation.map.chooseLocation geoLocation.map.chooseLocation(): Promise\<Location\>} to choose location on map.
+   *
+   * Fetches user location
    * @param props {@link LocationProps} - Specifying how the location request is handled
    * @param callback - Callback to invoke when current user location is fetched
    */
-  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void;
-  export function getLocation(
-    props: LocationProps,
-    callback?: (error: SdkError, location: Location) => void,
-  ): Promise<Location> {
-    ensureInitialized(FrameContexts.content, FrameContexts.task);
+  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void {
+    if (!callback) {
+      throw new Error('[location.getLocation] Callback cannot be null');
+    }
 
-    return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<Location>(getLocationHelper, callback, props);
-  }
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
 
-  function getLocationHelper(props: LocationProps): Promise<Location> {
-    return new Promise<Location>(resolve => {
-      if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
-        throw { errorCode: ErrorCode.OLD_PLATFORM };
-      }
-      if (!props) {
-        throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
-      }
-      if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
-      }
-      resolve(sendAndHandleError('location.getLocation', props));
-    });
+    if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
+      throw { errorCode: ErrorCode.OLD_PLATFORM };
+    }
+    if (!props) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+    }
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    sendMessageToParent('location.getLocation', [props], callback);
   }
 
-  /**
-   * Shows the location on map corresponding to the given coordinates
-   *
-   * @param location {@link Location} - which needs to be shown on map
-   * @returns Promise that will be fulfilled when the operation has completed
-   */
-  export function showLocation(location: Location): Promise<void>;
   /**
    * @deprecated
-   * As of 2.0.0, please use {@link location.showLocation location.showLocation(location: Location): Promise\<void\>} instead.
+   * As of 2.0.1, please use {@link geoLocation.map.showLocation geoLocation.map.showLocation(location: Location): Promise\<void\>} instead.
+   *
    * Shows the location on map corresponding to the given coordinates
+   *
    * @param location {@link Location} - which needs to be shown on map
```

Comment:
How about: `@param location - Location to be shown on the map` (Location type/link will be auto-rendered above this description)

Reply:
Thank you, Erika!! These comments are super helpful!!